.github/workflows/sonarcloud.yml

```
@@ -29,10 +31,18 @@ jobs:
+      - name: Run Tests and Analyze (All Microservices)
         env:
           SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
-        run: ./gradlew test jacocoTestReport sonarqube
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          for dir in project/backend/*/; do
+            if [ -f "$dir/build.gradle" ] && [ -f "$dir/gradlew" ]; then
+              echo "Running Sonar analysis for $dir"
+              (cd "$dir" && ./gradlew test jacocoTestReport sonarqube --info)
+            fi
+          done
```

YAMLlint (1.35.1)

[error] no new line character at the end of file

(new-line-at-end-of-file)